azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+        deleted_kv = self.app_config_client.delete_configuration_setting(
+            "not_exist_" + self.test_data.key_uuid
+        )
+        assert deleted_kv is None
+
+    def test_delete_correct_etag(self):
+        to_delete_kv = copy(self.test_data.no_label_data[-1])
+        to_delete_kv.key = "unit_test_key_" + self.test_data.key_uuid
+        to_delete_kv = self.app_config_client.add_configuration_setting(to_delete_kv)
+        deleted_kv = self.app_config_client.delete_configuration_setting(
+            to_delete_kv.key, etag=to_delete_kv.etag
+        )
+        assert deleted_kv is not None
+        with pytest.raises(CloudError):
+            self.app_config_client.get_configuration_setting(to_delete_kv.key)
+
+    def test_delete_wrong_etag(self):
+        to_delete_kv = copy(self.test_data.no_label_data[-1])
+        to_delete_kv.key = "unit_test_key_" + self.test_data.key_uuid
+        to_delete_kv = self.app_config_client.add_configuration_setting(to_delete_kv)
+        with pytest.raises(CloudError):
+            self.app_config_client.delete_configuration_setting(
+                to_delete_kv.key, etag="wrong etag"
+            )
+        self.app_config_client.delete_configuration_setting(to_delete_kv.key)
+
+    # method: list_configuration_settings
+    def test_list_configuration_settings_key_label(self):
+        to_list1 = self.test_data.label1_data[0]
+        to_list2 = self.test_data.label2_data[0]
+        items = self.app_config_client.list_configuration_settings(
+            labels=[to_list1.label, to_list2.label], keys=[to_list1.key, to_list2.key]
+        )
+        cnt = 0
+        for kv in items:
+            assert kv.key in [to_list1.key, to_list2.key] and kv.label in [
+                to_list1.label,
+                to_list2.label,
+            ]
+            cnt += 1
+        assert cnt == 2
+
+    def test_list_configuration_settings_only_label(self):
+        items = self.app_config_client.list_configuration_settings(
+            labels=[self.test_data.label1]
+        )
+        cnt = 0
+        for kv in items:
+            assert kv.label == self.test_data.label1
+            cnt += 1
+        assert cnt == 1
+
+    def test_list_configuration_settings_only_key(self):
+        to_list1 = self.test_data.no_label_data[0]
+        items = self.app_config_client.list_configuration_settings(keys=[to_list1.key])
+        cnt = 0
+        for kv in items:
+            assert kv.key == to_list1.key
+            cnt += 1
+        assert cnt >= 1
+
+    def test_list_configuration_settings_fields(self):
+        items = self.app_config_client.list_configuration_settings(
+            keys=["*"], labels=[self.test_data.label1], fields=["key", "content_type"]
+        )
+        cnt = 0
+        for kv in items:
+            assert (
+                kv.key is not None and kv.label is None and kv.content_type is not None
+            )
+            cnt += 1
+        assert cnt == 1
```

Comment:
if you are passing `key=*`, how can you guarantee that there is only going to be one result returned?

Reply:
self.test_data.label1 gurantees. There is only one ConfigurationSetting instance that has label == self.test_data.label1 in the service.